### PR TITLE
combine_bands Method for RasterLayers and TiledRasterLayers

### LINKED ISSUE #539: Expose the combineValues method to GeoPySpark
We should add a modified [combineValues](https://github.com/locationtech/geotrellis/blob/2419b48e7f3ab848c4dafe8cbd3e6fb2522ff298/spark/src/main/scala/geotrellis/spark/mapalgebra/CombineMethods.scala#L27) method to the layer classes. While it won't be as robust as the GeoTrellis version, we could still use it to do many useful things, For example, combining bands of one and `n` RDDs so that they produce an new value which has `n + 1` bands. This would be useful when working with datasets where each band of an image is its own file (ie. Landsat).

### PR DESCRIPTION
**Pull request by jbouffard**
This PR will allow the user to combine bands from different layers so that the resulting layer will have a `bandCount` equal to the sum of the `bandCount`s of the input layers. For example, if `layer a` has 2 bands and `layer b` has one, then the resulting layer will have a `bandCount` of 3. Where 2 of the bands come from `layer a` and 1 comes from `layer b`.

This PR resolves #539

### CODE REVIEW

File: geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TileLayer.scala [hunk 1]
```diff
@@ -239,4 +242,42 @@ object TileLayer {
...
+          bandMap + (value._1 -> value._2.bands),
+        (m1: Map[Int, Vector[Tile]], m2: Map[Int, Vector[Tile]]) =>
+          m1 ++ m2
+      )
+
+    bands.mapValues { case (v: Map[Int, Vector[Tile]]) =>
+      MultibandTile(
+        v.toSeq
+          .sortWith(_._1 < _._1)
+          .toMap
```
Comment (echeipesh) [inline geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TileLayer.scala]: I believe turning the collection back to map may undo the sort and isn't actually required step here.

File: geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TileLayer.scala [hunk 2]
```diff
@@ -239,4 +242,42 @@ object TileLayer {
...
+    val arr = Array.ofDim[RDD[(K, (Int, MultibandTile))]](rdds.size)
+
+    for ((layer, index) <- rdds.zipWithIndex) {
+      arr(index) = layer.mapValues { (index, _) }
+    }
+
+    val unioned = sc.union(arr.toSeq)
+
+    val bands: RDD[(K, Map[Int, Vector[Tile]])] =
+      unioned.combineByKey(
```
Comment (echeipesh) [inline geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TileLayer.scala]: There should be some kind of check for error or warning if the layouts of the different layers do not match. Thats a kind of error thats easy to do with a surprising result.

File: geopyspark/geotrellis/union.py [hunk 3]
```diff
@@ -26,37 +26,16 @@ def union(*layers):
     """
 
     if len(layers) == 1:
-        raise ValueError("Union can only performed on 2 or more layers")
+        raise ValueError("combine_bands can only performed on 2 or more layers")
```
Comment (echeipesh) [inline geopyspark/geotrellis/union.py]: Search/replace error ?

File: geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TileLayer.scala (hunk 2 above)
Comment (jbouffard) [reply geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TileLayer.scala]: I had the layouts checked in python: https://github.com/jbouffard/geopyspark/blob/acb4163cbbcfdd8ad6dc3adceef5a9c6bf96effc/geopyspark/geotrellis/__init__.py#L66

File: geopyspark/geotrellis/union.py (hunk 3 above)
Comment (jbouffard) [reply geopyspark/geotrellis/union.py]: Yeah, that's definitely a search/replace error.

File: geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TileLayer.scala (hunk 1 above)
Comment (jbouffard) [reply geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TileLayer.scala]: Ah, I didn't know that. I'll fix it.

File: geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TileLayer.scala (hunk 2 above)
Comment (echeipesh) [reply geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TileLayer.scala]: Gotcha

**Pull request merged**